application/controllers/admin/DataEntry.php

```diff
@@ -1884,20 +1884,11 @@ public function insert()
                     }
                     // query for updating tokens uses left
                     $aToken = Token::model($surveyid)->findByAttributes(['token' => $_POST['token']]);
+                    $aToken->usesleft = ((int) $aToken->usesleft) - 1;
                     if (isTokenCompletedDatestamped($thissurvey)) {
-                        if ($aToken->usesleft <= 1) {
-                            $aToken->usesleft = ((int) $aToken->usesleft) - 1;
-                            $aToken->completed = $submitdate;
-                        } else {
-                            $aToken->usesleft = ((int) $aToken->usesleft) - 1;
-                        }
+                        $aToken->completed = $submitdate;
                     } else {
-                        if ($aToken->usesleft <= 1) {
-                            $aToken->usesleft = ((int) $aToken->usesleft) - 1;
-                            $aToken->completed = 'Y';
-                        } else {
-                            $aToken->usesleft = ((int) $aToken->usesleft) - 1;
-                        }
+                        $aToken->completed = 'Y';
                     }
```

Comment:
just had another look and you need to make another adjustment, it can only be substracted after the condition was checked, so you need to add the substraction behind the enitre condition block here "$aToken->usesleft = ((int) $aToken->usesleft) - 1;"